perf/lua/1mops_write.lua

```diff
@@ -0,0 +1,325 @@
+      num_ops, ops_per_txn, num_fibers, nodes, index_config.type, hints_msg))
+
+box.cfg{
+    log_level = 0,
+    listen = 0,
+    read_only = false,
+    memtx_memory = 2*1024*1024*1024,
+    work_dir=test_dir,
+    wal_mode='write',
+}
+
+box.schema.user.create('replicator', {password = 'password'})
+box.schema.user.grant('replicator', 'replication')
+
+-- number of nodes, storage for popen handles
+local nodes_ph = {}
+
+-- run replicas (if needed)
+if (nodes > 1) then
+    if (nodes ~= nil and nodes < 32 and nodes > 0) then
+        if (type(popen) ~= 'table' or popen.new == nil) then
+            exit(1, 'can\'t run replication w/o popen')
+        end
+        print('# starting ' .. nodes - 1 .. ' replicas')
+        for i = 3302, 3300+nodes do
+            -- subdir for replica's data
+            os.execute('mkdir ' .. i)
+            -- command line for replica to start
+            local cmd = {
+                arg[-1],
+                '-e',
+                string.format([[
+                box.cfg {
+                    read_only = false,
+                    log_level = 5,
+                    replication = {'replicator:password@%s'},
+                    work_dir = '%s',
+                    replication_connect_quorum=1
+                }]], box.info.listen, i)
+            }
+            local res, err = popen.new(cmd, std_redirect)
+            if (res) then
+                nodes_ph[i] = res
+            else
+                exit(1, 'error running replica: ' .. err)
+            end
+        end
+
+        -- wait for all replicas to connect
+        while #box.info.replication < nodes do
+            print('# replication', #box.info.replication)
+            require('fiber').sleep(0.1)
+        end
+    else
+        exit(1, 'Incorrect number of nodes \"' .. arg[1] .. '\" must be 1..31')
+    end
+end
+
+local space
+local done = false
+local err
+
+if (test_qsync) then
+    box.cfg{replication_synchro_quorum = nodes}
+    print('# promoting')
+    box.ctl.promote()
+    print('# done')
+    space, err = box.schema.create_space('test',
+                                         {engine = engine, is_sync = true})
+else
+    space, err = box.schema.create_space('test',
+                                         {engine = engine})
+end
+if space == false then
+    exit(1, 'error creating space ' .. json.encode(err))
+end
+local res, err = pcall(space.create_index, space, 'pk', index_config)
+if (res ~= true) then
+    exit(1, 'error creating index ' ..
+         json.encode(index_config) .. ' :' .. json.encode(err))
+end
+
+-- THE load fiber
+local function fiber_load(start, s)
+    start = start % 1000000 -- limit the size of space to 1M elements
+    for _ = 1, trans_per_fiber do
+        box.begin()
+        for _ = 1, ops_per_txn do
+            s:replace{start}
+            start = start + 1
+        end
+        box.commit()
+    end
+end
+
+-- fiber storage to join them
+local fibers_storage = {}
+
+collectgarbage()
```

Comment:
Nit: It also would be nice to have an option whether run in a "dirty"/"clean" environment or even with GC disabled.

Reply:
I believe it should be eliminated as a class of possible fluctuation - these GC triggers were introduced to stabilize the results, still they failed. I would like to keep them as is, proceeding with further search of the fluctuation source.
Overloading the test with big number of options is no good IMHO.